### ISSUE TITLE
Lint/AssignmentInCondition false positive

### ISSUE THREAD

**Issue by davidhrbac** [rubocop-hq/rubocop#5253]
```
logins_to_create = {}
# loop over these numbers
(serial_from..serial_to).each do |serial|
  logins_to_create[format('%s-%02d', pid_dc, serial).to_s] = true
end
```
## Expected behavior

Should not trigger the warning. We are just setting the variable to `true`.

## Actual behavior

```
app/controllers/user_projects_controller.rb:139:75: W: Lint/AssignmentInCondition: Use == if you meant to do a comparison or wrap the expression in parentheses to indicate you meant to assign in a condition. (https://github.com/bbatsov/ruby-style-guide#safe-assignment-in-condition)
                 logins_to_create[format('%s-%02d', pid_dc, serial).to_s] = true
                                                                          ^
```

## RuboCop version

```
$ rubocop -V
0.52.0 (using Parser 2.4.0.2, running on ruby 2.4.2 x86_64-linux)
```

**Comment by pocke** [comment]
Can you give me more reproduce code? I can't reproduce it.

Put it as `test.rb`

```ruby
logins_to_create = {}
# loop over these numbers
(serial_from..serial_to).each do |serial|
  logins_to_create[format('%s-%02d', pid_dc, serial).to_s] = true
end
```

Run rubocop

```bash
$ rubocop 
Inspecting 1 file
C

Offenses:

test.rb:4:28: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
  logins_to_create[format('%s-%02d', pid_dc, serial).to_s] = true
                           ^^

1 file inspected, 1 offense detected


$ rubocop -V
0.52.0 (using Parser 2.4.0.2, running on ruby 2.4.2 x86_64-linux)
```

**Comment by davidhrbac** [comment]
```
 $ cat test.rb 
logins_to_create = {}
# loop over these numbers
(serial_from..serial_to).each do |serial|
  logins_to_create[format('%s-%02d', pid_dc, serial).to_s] = true
end
✔ ~/scs-ror [gem_update L|…⚑ ] 
16:02 $ rubocop test.rb 
Inspecting 1 file
C

Offenses:

test.rb:4:28: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
  logins_to_create[format('%s-%02d', pid_dc, serial).to_s] = true
                           ^^

1 file inspected, 1 offense detected
```

This is OK.

Running on my code gives:
```
rubocop app/controllers/user_projects_controller.rb
Inspecting 1 file
W

Offenses:

app/controllers/user_projects_controller.rb:141:75: W: Lint/AssignmentInCondition: Use == if you meant to do a comparison or wrap the expression in parentheses to indicate you meant to assign in a condition. (https://github.com/bbatsov/ruby-style-guide#safe-assignment-in-condition)
                 logins_to_create[format('%s-%02d', pid_dc, serial).to_s] = true
                                                                          ^

1 file inspected, 1 offense detected
```

I have created snippet:

```ruby
          if serial_from >= 0 && serial_from <= 99 &&
             serial_to >= 0 && serial_to <= 99 &&

             # are boundaries in right order?
             if serial_from <= serial_to

               # loop over these numbers
               (serial_from..serial_to).each do |serial|
                 # rubocop:disable Style/FormatStringToken
                 # FIXME https://github.com/bbatsov/rubocop/issues/5245
                 logins_to_create[format('%s-%02d', pid_dc, serial).to_s] = true
                 # rubocop:enable Style/FormatStringToken
                 # FIXME https://github.com/bbatsov/rubocop/issues/5245
               end
             else
               # nonsense request
               login_error << t + " contains boundaries in wrong order: from(" +
                              serial_from.to_s + ") is greater than to(" + serial_to.to_s + "), request has no sense."

             end

          # otherwise we check that login match correct format of our project..
          # login match name of our pid exactly?
          # ie. dd-14-9 means to test t_rest for value "" (empty string)
          elsif t_rest.empty?
            # this we do not allow
            # (otherwise uncomment: logins_to_create[t] = true )
            login_error << t + " is not allowed to use exactly same login as is your pid " +
                           pid + ", correct format is " + pid_dc + "-NN, where NN is number."

          # login match name of our pid with two digit appendix? (ie. dd-14-9-02)
          # it means single member from serie,
          # ie dd-14-9-02 we test like a t_rest for value "-02"
          elsif t_rest.match?(/\A\-\d\d\Z/)
            logins_to_create[t] = true

          # otherwise is appended something incorrect to my pid (ie. dd-14-9-test)
          else
            login_error << t + " is not allowed format of login for your pid " +
                           pid + ", correct format is " + pid_dc + "-NN, where NN is number."
          end
```
```
rubocop test.rb 
Inspecting 1 file
W

Offenses:

test.rb:1:11: C: Layout/InitialIndentation: Indentation of first line in file detected.
          if serial_from >= 0 && serial_from <= 99 &&
          ^^
test.rb:11:75: W: Lint/AssignmentInCondition: Use == if you meant to do a comparison or wrap the expression in parentheses to indicate you meant to assign in a condition. (https://github.com/bbatsov/ruby-style-guide#safe-assignment-in-condition)
                 logins_to_create[format('%s-%02d', pid_dc, serial).to_s] = true
                                                                          ^
test.rb:42:1: C: Layout/TrailingBlankLines: 1 trailing blank lines detected. (https://github.com/bbatsov/ruby-style-guide#newline-eof)

1 file inspected, 3 offenses detected
```

**Comment by pocke** [comment]
Thank you! I can reproduce the problem, and I can create a minimum reproduce code.

```ruby
if cond &&
    if cond2
      foo[bar] = baz
    end
end
```

**Comment by davidhrbac** [comment]
Thanks!

**Comment by Drenmi** [comment]
> Thank you! I can reproduce the problem, and I can create a minimum reproduce code.

That is ... valid Ruby? 😱

**Comment by davidhrbac** [comment]
Well, the code is not beautiful and you can see just the part of it. :angry: Anyway, the author is not member of the project anymore.

**Comment by Drenmi** [comment]
My fear is that we'd need to account for this kind of code in a lot of existing cops. It is probably better to introduce a cop that prevents the adding of such code in the first place. WDYT?

**Comment by davidhrbac** [comment]
Syntax is not pure but the code is valid, Ruby is OK with that. I'd correct this false positive. Then we can introduce the cop to rule out code like this.
